frontend/src/components/HeaderUser.tsx

```diff
@@ -0,0 +1,58 @@
+import "../index.css";
+import React, { useState } from "react";
+import { SNOWPACK_PUBLIC_API_URL } from "../model";
+
+const logout = async () => {
+  await fetch(`${SNOWPACK_PUBLIC_API_URL}/api/logout`, {
+    method: "POST",
+  })
+    .then((res) => {
+      if (res.ok) {
+        return true;
+      } else {
+        throw new Error("Could not log out.");
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+      return false;
+    });
+  // Redirect to login page
+  window.location.href = "/";
+};
+
+const showSettings = () => {
+  return (
+    <div class="settings-list" aria-labelledby="dropdownMenuLink">
+      <a class="dropdown-item" onClick={logout}>
+        Log out
+      </a>
+    </div>
+  );
+};
+
+export const HeaderUser = ({ username }: { username: string }) => {
+  const [userSettings, setUserSettings] = useState<boolean>(false);
```

Comment:
This is a thought / suggestion:
With a variable called `userSettings` I expect it to contain actual settings, e.g. in an array or object. 
The boolean one you declared I would expect to be called `showUserSettings` 
Change if you agree :)